src/lib/support/PrivateHeap.h

```diff
@@ -38,6 +38,12 @@ void * PrivateHeapAlloc(void * heap, size_t size);
 // Marks the specified block as free
 void PrivateHeapFree(void * ptr);
 
+// Reallocate a block to a new size
+// Special arguments:
+//    - null ptr value represents an Alloc
+//    - zero size represents a Free
```

Comment:
Are we sure we want to support these special usages when standard implementations do not?

Reply:
I took these from the realloc documentation from Linux:

```
If ptr is NULL, then the call is equivalent to malloc(size), for all values of size; if size is equal to zero, and ptr is not NULL,
       then the call is equivalent to free(ptr).Y
       ```

Reply:
Sorry, I should have said that nat all standard implementations support these.

This is fine.  Not a big deal.  It's clearly defined in the comments.  Looks good.